### PR TITLE
GP-30260 GreenpeaceMessageTemplate.send API

### PR DESCRIPTION
**Pull request by viliam-ivanov**
Wrapper for MessageTemplate.send -> GreenpeaceMessageTemplate.send API accepts two params:
id (message template id)
to_emails (comma separated list of emails)

### CODE REVIEW

File: api/v3/GreenpeaceMessageTemplate/Send.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+ */
+function _civicrm_api3_greenpeace_message_template_send_spec(&$spec) {
+  $spec['id'] = [
+    'name'         => 'id',
+    'title'        => 'MessageTemplate ID',
+    'type'         => CRM_Utils_Type::T_INT,
+    'api.required' => 1
+  ];
+
+  $spec['to_emails'] = [
```
Comment (pfigel) [inline api/v3/GreenpeaceMessageTemplate/Send.php]: Let's keep the param name as `to_email` (singular) so we don't have to change the parameter name in all tasks.

File: api/v3/GreenpeaceMessageTemplate/Send.php [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    'invalid' => []
+  ];
+
+  try {
+    $emails = explode(",", $params['to_emails']);
+    foreach ($emails as $email) {
+      if (!filter_var($email, FILTER_VALIDATE_EMAIL)) {
+        $return_values['invalid'][] = $email;
+      } else {
+        $message_template = civicrm_api3('MessageTemplate', 'send', [
```
Comment (pfigel) [inline api/v3/GreenpeaceMessageTemplate/Send.php]: We can pass through `$params` as a whole (except for `to_email`, which we set to `$email`) so we support all other parameters supported by `MessageTemplate.send` as well. (They don't have to be defined in `_civicrm_api3_greenpeace_message_template_send_spec`, they'll just magically work if someone provides values). For example, we sometimes also use `template_params`.

**Review by pfigel — CHANGES_REQUESTED** [review]
Two small issues, otherwise LGTM.

File: api/v3/GreenpeaceMessageTemplate/Send.php (hunk 1 above)
Comment (viliam-ivanov) [reply api/v3/GreenpeaceMessageTemplate/Send.php]: commit e38c4661d58158066094fc5a5c371e41a48b8706

File: api/v3/GreenpeaceMessageTemplate/Send.php (hunk 2 above)
Comment (viliam-ivanov) [reply api/v3/GreenpeaceMessageTemplate/Send.php]: commit e38c4661d58158066094fc5a5c371e41a48b8706

**Review by pfigel — APPROVED** [review]
💯 :shipit:

**Pull request merged**